app/models/katello/host/subscription_facet.rb

```diff
@@ -298,11 +298,12 @@ def backend_update_needed?
       end
 
       def self.populate_fields_from_facts(host, parser, _type, _source_proxy)
+        return unless host.subscription_facet # skip method if the host is not a registered host
         has_convert2rhel = parser.facts.key?('conversions.env.CONVERT2RHEL_THROUGH_FOREMAN')
```

Comment:
Would the following also work?
```suggestion
        has_convert2rhel = parser.facts.key?('conversions.env.CONVERT2RHEL_THROUGH_FOREMAN')
        return unless (host.subscription_facet  || has_convert2rhel) # skip method if the host is not a registered host
```

That'd create the facet if the host has the special fact, and skip otherwise?

Reply:
I tried that in https://github.com/Katello/katello/pull/11110 and ran a pipeline based on the packit build -- it passed.

Reply:
I think it's an interesting thing to think about. AFAIK `populate_fields_from_facts` is called for every fact source, so also Puppet. That makes me think the current code will result in unstable values if both RHSM and Puppet are used. Your code to skip if no fact exists is probably more reliable, but also means no mechanism exists to clear the value.